docs/Theory_cluster_batch_scheduling.rst

```diff
@@ -1,15 +1,132 @@
-"place" is in fact one of many computers synchronised to the manager
-which are called :ref:`The compute nodes`.
-
-In fact it's a bit trickier than that, but we'll stay at this abstraction level
-for now.
-
-.. include:: Theory_cluster_slurm.rst
+On a cluster, users don't have direct access to the compute nodes but
+instead connect to a login node and add jobs to the workload manager
+queue. Whenever there are resources available to execute these jobs
+they will be allocated to a compute node and run, which can be
+immediatly or wait for several days.
+
+
+Anatomy of a job
+----------------
+
+A job is comprised of a number of steps that will run one after the
+other. This is done so that you can schedule a sequence of processes
+that can use the results of the previous steps without haveing to
+manually interact with the scheduler.
+
+Each step can have any number of task which are groups of processes
+that can be scheduled independently on the cluster but can run in
+parallel if there are resources available. The distinction between
+steps and tasks is that multiple tasks, if they are part of the same
+step, cannot depend on results of other task because there are no
+guarantees on the order in which they will be executed.
+
+Finally each process group is the basic unit that is scheduled in the
+cluster. It comprises of a set of processes (or threads) that can run
+on a number of resources (CPU, GPU, RAM, ...) and are scheduled
+together as a unit on one or more machines.
+
+Each of these concepts lends itself to a particular use. For multi-gpu
+training in AI workloads you would use one task per GPU for data
+paralellism or one process group if you are doing model
+parallelism. Hyperparameter optimisation can be done using a
+combination of tasks and steps but is probably better left to a
+framework outside of the scope of the workload manager.
+
+If this all seems complicated, you should know that all these things
+do not need to always be used. It is perfectly acceptable to sumbit
+jobs with a single step, a single task and a single process.
+
+
+Understanding the queue
+-----------------------
+
+The available resources on the cluster are not infinite and it is the
+workload manager's job to allocate them. Whenever a job request comes
+in and there are not enough resources available to start it
+immediately, it will go in the queue.
+
+Once a job is in the queue, it will stay there until another job
+finishes and then the workload manager will try to use the newly freed
+resources with jobs from the queue. The exact order in which the jobs
+will start is not fixed, because it depends on the local policies
+which can take into account the user priority, the time since the job
+was requested, the amount of resources requested and possibly other
+things. There should be a tool that comes with the manager where you
+can see the status of your queued jobs and why they remain in the
+queue.
```

Comment:
@btravouillon , @achillemyette1 , should we add a section on that tool?

Reply:
There is one already for Slurm squeue: https://docs.mila.quebec/Userguide.html#checking-job-status

Reply:
I don't think so. Your writing is Slurm-agnostic, I believe we should keep it as-is.